### ISSUE TITLE
Py3: Segfault on close

### ISSUE THREAD

**Issue by DmitryKuzmenko** [hercules-team/python-augeas#30]
The following code works on Python2 but segfaults on Python3
```python
import augeas

if __name__ == '__main__':
    aug = augeas.Augeas()
    aug.close()
```
It happens here: https://github.com/hercules-team/python-augeas/blob/master/augeas/__init__.py#L578

Environment:
```bash
$ python2 --version
Python 2.7.13
$ python3 --version
Python 3.6.1
$ pacman -Ss augeas | grep "^\S"
community/augeas 1.8.0-1 [installed]
community/python-augeas 1.0.2-2 [installed]
community/python2-augeas 0.5.0-1 [installed]
```
BTW, the following code works good:
```c
#include <augeas.h>

int main(int argc, char **argv) {
        augeas *aug;
        aug = aug_init(NULL, NULL, 0);
        aug_close(aug);
}
```

**Comment by bmw** [comment]
This is a problem with python-augeas 1.0.2 (and/or it's dependencies). I think the reason you didn't hit this problem with Python 2 is you were still using python-augeas 0.5.0.

I've reproduced this problem on Ubuntu 16.04 using Python 2.7.12:
```
$ dpkg -l *augeas*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                   Version                  Architecture             Description
+++-======================================-========================-========================-==================================================================================
un  augeas-doc                             <none>                   <none>                   (no description available)
ii  augeas-lenses                          1.4.0-0ubuntu1           all                      Set of lenses needed by libaugeas0 to parse config files
ii  augeas-tools                           1.4.0-0ubuntu1           amd64                    Augeas command line tools
ii  libaugeas0                             1.4.0-0ubuntu1           amd64                    Augeas configuration editing library and API
$ virtualenv venv
Running virtualenv with interpreter /usr/bin/python2
New python executable in /home/bmw/Development/certbot/certbot/venv/bin/python2
Also creating executable in /home/bmw/Development/certbot/certbot/venv/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
$ . ./venv//bin/activate
$ pip install python-augeas
Collecting python-augeas
Collecting cffi>=1.0.0 (from python-augeas)
  Using cached cffi-1.10.0-cp27-cp27mu-manylinux1_x86_64.whl
Collecting pycparser (from cffi>=1.0.0->python-augeas)
Installing collected packages: pycparser, cffi, python-augeas
Successfully installed cffi-1.10.0 pycparser-2.17 python-augeas-1.0.2
$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import augeas
>>> a = augeas.Augeas()
>>> a.close()
Segmentation fault (core dumped)
```
Additionally, because the same problematic function is called [automatically during garbage collection](https://github.com/hercules-team/python-augeas/blob/b63faa87d7a19d72dafdc960b1af2b5241ec3b74/augeas/__init__.py#L108) this can cause programs using `python-augeas` to segfault even if they avoid the problematic `close()` function.

This problem makes `python-augeas` unusable in programs that don't keep the `Augeas` object for the entire lifetime of the program.

**Comment by thedrow** [comment]
I wonder what could be the problem. I'll try to check if the handle is a null pointer as well as checking if it's none.

**Comment by thedrow** [comment]
Should be fixed in 1.0.3 which is being released now :)

**Comment by FelixSchwarz** [comment]
I think I found a bug in commit bd62d09 as `.close()` is not called anymore automatically. I created a PR in #55. I verified that certbot's test suite still passes so I guess I did not break anything major by re-enabling the `close()` method.
